### PR TITLE
Migrate `Event` to `zod`

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This migrates the validation of `Event` to `zod`:

- `ChainUpdate`
- `DeletedMultisigTransaction`
- `ExecutedTransaction`
- `IncomingEther`
- `IncomingToken`
- `MessageCreated`
- `ModuleTransaction`
- `NewConfirmation`
- `NewMessageConfirmation`
- `OutgoingEther`
- `OutgoingToken`
- `PeningTransaction`
- `SafeAppsUpdate`

## Changes

- Remove `EventValidationPipe` and associated schema.
- Add `WebHookSchema`, constituted of schemas for all above listed events and infer types from them.
- Update associated tests.

### CODE REVIEW

File: src/routes/cache-hooks/cache-hooks.controller.ts [hunk 1]
```diff
@@ -17,7 +18,9 @@ export class CacheHooksController {
   @UseGuards(PreExecutionLogGuard, BasicAuthGuard)
   @Post('/hooks/events')
   @HttpCode(202)
-  async postEvent(@Body(EventValidationPipe) event: Event): Promise<void> {
+  async postEvent(
+    @Body(new ValidationPipe(WebHookSchema)) event: Event,
```
Comment (fmrsabino) [inline src/routes/cache-hooks/cache-hooks.controller.ts:22]: What is the response code if the we receive an event that we don't know how to handle?

Comment (fmrsabino) [reply src/routes/cache-hooks/cache-hooks.controller.ts:22]: Nevermind, forgot that we had a global exception filter:

https://github.com/safe-global/safe-client-gateway/blob/ba418d0cd54e5074384a8710ff9176ea11f9ca86/src/routes/common/filters/zod-error.filter.ts#L25-L39

So the expected status code would be 422 which I would say is correct for this route 👍

File: src/routes/cache-hooks/entities/schemas/__tests__/chain-update.schema.spec.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          code: 'invalid_literal',
+          expected: 'CHAIN_UPDATE',
+          path: ['type'],
+          message: 'Invalid literal value, expected "CHAIN_UPDATE"',
+        },
+      ]),
+    );
+  });
+
+  it.each([['type' as const], ['chainId' as const]])(
```
Comment (fmrsabino) [inline src/routes/cache-hooks/entities/schemas/__tests__/chain-update.schema.spec.ts:36]: Do we need to use tuples in this test (over the actual string values)?

**Review by fmrsabino — APPROVED** [review]

File: src/routes/cache-hooks/entities/schemas/__tests__/chain-update.schema.spec.ts (hunk 2 above)
Comment (iamacook) [reply src/routes/cache-hooks/entities/schemas/__tests__/chain-update.schema.spec.ts]: Removed in 90ae30c716907fc58d03eb6231abe3ac7eb6e304.

**Review by fmrsabino — APPROVED** [review]
🚀 👏

**Pull request merged**